crates/sui-graphql-rpc/src/types/event.rs

```diff
@@ -2,28 +2,67 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use async_graphql::*;
+use sui_indexer::models_v2::events::StoredEvent;
+
+use crate::error::Error;
 
 use super::{
-    address::Address, base64::Base64, date_time::DateTime, move_module::MoveModuleId,
-    move_type::MoveType, sui_address::SuiAddress,
+    address::Address, date_time::DateTime, move_module::MoveModuleId, move_value::MoveValue,
+    sui_address::SuiAddress,
 };
 
 #[derive(SimpleObject)]
+#[graphql(complex)]
 pub(crate) struct Event {
-    /// Package id and module name of Move module that the event was emitted in
-    pub sending_module_id: Option<MoveModuleId>,
-    /// Package, module, and type of the event
-    pub event_type: Option<MoveType>,
-    pub senders: Option<Vec<Address>>,
+    #[graphql(skip)]
+    pub stored: StoredEvent,
+    #[graphql(flatten)]
+    pub contents: MoveValue,
+}
+
+#[ComplexObject]
+impl Event {
+    /// Package id and module name of the Move module that the event was emitted in
+    async fn sending_module_id(&self) -> Result<Option<MoveModuleId>> {
+        let package_id = SuiAddress::from_bytes(&self.stored.package)
+            .map_err(|e| Error::Internal(e.to_string()))
+            .extend()?;
+        Ok(Some(MoveModuleId {
+            package: package_id,
+            name: self.stored.module.clone(),
+        }))
+    }
+
+    /// Addresses of the senders of the event
+    async fn senders(&self) -> Result<Option<Vec<Address>>> {
+        let result: Result<Option<Vec<Address>>, _> = self
+            .stored
+            .senders
+            .iter()
+            .map(|sender| {
+                sender
+                    .as_ref()
+                    .map(|sender| {
+                        SuiAddress::from_bytes(sender)
+                            .map(|sui_address| Address {
+                                address: sui_address,
+                            })
+                            .map_err(|e| Error::Internal(e.to_string()))
+                    })
+                    .transpose()
+            })
+            .collect();
+
+        result.extend()
```

Comment:
I see, I think it's better to filter out the None's/ prioritize returning partial instead of no response

Reply:
That's my intuition as well, but could you double check with data platform on what would cause one of these senders to be `None`?

Reply:
postgres allows notnull array columns, but does not have a built-in typing for non-nullable arrays with non-nullable elements, so unfortunately it will type these columns w/ nullable elements. diesel schema mirrors postgres so it picks up this quirk. indexer has logic to complain should the senders array contain None